### PR TITLE
Localize code style option previews

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes part 3 of #9674

## Ask Mode Info

**Customer scenario**: The previews for Code Styles were not localized, making it hard for non-English users to understand which choice resulted in which code.
**Bugs this fixes**: Part 3 of #9674
**Workarounds**: None
**Risk**: Very low
**Performance impact**: None.
**Is this a regression from a previous update?**: No, this has been broken since the UI was introduced
**Root cause analysis**: It was skipped as part of the original feature work.
**How was the bug found?**: Ad hoc testing (see screenshot from #9674)

### CODE REVIEW

**Comment by dpoeschl** [comment]
Tagging for review @dotnet/roslyn-ide @CyrusNajmabadi @nguerrera

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs [hunk 1]
```diff
@@ -22,576 +22,574 @@ internal class StyleViewModel : AbstractOptionPreviewViewModel
...
 class Program
-{
+{{
 //[
-    private Int32 _member;
-    static void M(Int32 argument)
-    {
-        Int32 local;
-    }
+    private Int32 {ServicesVSResources.Underscore_member};
```
Comment (rchande) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: Is it worth using these names that need to be localized in lieu of single-letter identifiers like `M` which you haven't localized?

**Comment by rchande** [comment]
👍

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs (hunk 1 above)
Comment (dpoeschl) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: I decided that I didn't want to change the previews in any way, so I did this. And I didn't think loc'ing single letter identifiers was necessary, but that's why @nguerrera is on the review.

**Comment by dpoeschl** [comment]
@nguerrera Can you comment on the single letter variable names and anything else you see?

**Comment by dpoeschl** [comment]
Tagging @MattGertz for Ask Mode consideration. I have two signoffs and the Ask Mode Template is filled out at the top, but I still want @nguerrera's opinion before merging.

**Comment by MattGertz** [comment]
Whenever you are comfortable with this.

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs [hunk 2]
```diff
@@ -22,576 +22,574 @@ internal class StyleViewModel : AbstractOptionPreviewViewModel
...
+}}";
 
-        private static readonly string s_propertyDeclarationPreviewTrue = @"
-class C{
-    public int Id { get; set; }
+        private static readonly string s_propertyDeclarationPreviewTrue = $@"
+class C{{
+    public int {ServicesVSResources.Id} {{ get; set; }}
     void Method()
-    {
```
Comment (nguerrera) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: In some places you've kept Method, but elsewhere renamed to M. Is this deliberate and if so, what's the rationale?

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs (hunk 1 above)
Comment (nguerrera) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: I think you could make actually make a case that you don't need to localize any member of parameter names. Any user of .NET would be very used to reading these in English only as none of the API they use daily are localized.

Comment (nguerrera) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: Along those lines, I think that localizing the comments alone might have been sufficient. 

Regardless, I don't think we should use one-letter identifiers just to side-step the issue.

**Comment by dpoeschl** [comment]
Talked to @nguerrera about this and we feel good about the change.

**Comment by dpoeschl** [comment]
@nguerrera 
> Along those lines, I think that localizing the comments alone might have been sufficient.

Even leaving things like Customer and Age unloc'd?

**Comment by nguerrera** [comment]
> Even leaving things like Customer and Age unloc'd?

Maybe we should use less domain specific names. This actually raises a new worry for me: that they might be translated with a formatting that invalidates the style we're trying to demonstrate, or even in to invalid identifiers (e.g. more than one word separated by a space).

MSDN samples are not localized: e.g. https://msdn.microsoft.com/fr-fr/library/b3h1hf19(v=vs.110).aspx

I therefore think it is fine to not localize the *code* in these previews. I do think that localizing the comments would be nice and using such comments to help user's who may not speak english understand what is significant about the style of these identifiers in the given context.

It's not necessarily wrong to localize the identifiers in these previews, but it is going beyond what we've done elsewhere, and does carry some risk of incorrect translation.

**Comment by dpoeschl** [comment]
Okay, going to change this to *only* translate the comments.

**Comment by dpoeschl** [comment]
@nguerrera @rchande @dotnet/roslyn-ide I've redone this to *only* localize the comments in code, since this more closely matches how we do things on MSDN and reduces the chance that a single-word identifier will be localized into a multiple-word phrase in some other language, which would break the code.

**Comment by dpoeschl** [comment]
And I realize I missed a bunch of comments. Stay tuned.

**Comment by dpoeschl** [comment]
Okay, good to go.

**Comment by dpoeschl** [comment]
@MattGertz You approved this for a previous release, but it's finally ready now. Is it still approved for `2.0 (RC.3)`?

**Comment by MattGertz** [comment]
Yup.

**Pull request merged**